### ISSUE TITLE
Import ignores from .gitignore

### ISSUE THREAD

**Issue by sandstrom** [adrienverge/yamllint#360]
## Problem

Right now there is a lot of duplication between our yaml-lint ignore directives, and our .gitignore files. It would be neat if we could import from gitignore, via e.g. `import_ignore: .gitignore`.

## How it could work

Something like:

```yaml
# using the default rules
extends: default

# skip everything ignored by .gitignore
import_ignore: ".gitignore"

rules:
  octal-values: enable
  # …
```

Since I think yaml-lint is using the same ignore format as .gitignore, maybe it's fairly easy to import? Would also allow keeping the ignore list in a separate file.

## Open Questions
- How should conflicting rules be solved? Ideally with "last rule wins". That way one can override or adjust directives imported from .gitignore if needed.

**Comment by adrienverge** [comment]
I understand the use-case. It could be done using a conf option `ignore-from-file: .gitignore`, mutually exclusive with [`ignore`](https://yamllint.readthedocs.io/en/stable/configuration.html#ignoring-paths). Do you want to implement it?

**Comment by sandstrom** [comment]
@adrienverge Mutually exclusive would preclude making adjustments to the ignore directives, or overriding `!keep-me.file`. Are there technical reasons making it more complicated to allow mixing `ignore` with `import_ignore`?

I'd love to, but I haven't spent many hours in Python. So not sure I could.

**Comment by andrewnaguib** [comment]
@adrienverge I'm willing to implement that. Would you assign me and give me few tips? :))

**Comment by adrienverge** [comment]
> @adrienverge Mutually exclusive would preclude making adjustments to the ignore directives, or overriding `!keep-me.file`. Are there technical reasons making it more complicated to allow mixing `ignore` with `import_ignore`?

Being able to make adjustments makes sense, I'm not opposed to it. It's just that it's more complex to implement: more code and more tests are needed.

About the option name, I'd prefer staying consistent with other configuration keys. For example something like `ignore-from-file: .gitignore`, or even better: `ignore-from-files: [.gitignore, other-file]`.

> @adrienverge I'm willing to implement that. Would you assign me and give me few tips? :))

Great! I suggest looking carefully at a few commits that implemented new features, and reproduce the same. Note that they need documentation, test every corner case, and have a commit message that describes what they do. Please save my time and open the PR once it's 100% ready.
